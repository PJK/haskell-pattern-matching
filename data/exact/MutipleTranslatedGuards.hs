module MutipleTranslatedGuards where

succ :: Int -> Int -> Bool
succ 0 1 = True
succ 1 2 = True
succ 2 3 = True
succ _ _ = False