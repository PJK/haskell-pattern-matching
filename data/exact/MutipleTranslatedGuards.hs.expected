{
    "tag": "AnalysisSuccess",
    "contents": [
        [],
        [
            [
                "succ",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
